pandas/core/indexes/base.py

```diff
@@ -1832,15 +1832,31 @@ def __contains__(self, key):
             return False
 
     _index_shared_docs['contains'] = """
-        return a boolean if this key is IN the index
+        Return a boolean if key is in the index.
+
+        This function returns True if key is in the index and
+        returns False if key is not in index.
 
         Parameters
         ----------
-        key : object
+        key : immutable object
+            Key to be searched.
 
         Returns
         -------
-        boolean
+        bool
+            Result of the key search.
+
+        Notes
+        -----
+            CategoricalIndex is a sub-class of Index.
```

Comment:
Don't think you need this Notes section.

Reply:
Could you add a See Also with Index.isin

Reply:
this note is because of shared documentation between Index and CategoricalIndex